### PR TITLE
Fall back to "unknown" if cluster name is empty

### LINKED ISSUE #32: CouchbaseClusterSample not reporting - Error and <empty> entity name
#### CouchbaseClusterSample Not Reporting 

There are no values being reported for the CouchbaseClusterSample.

##### Configuration
```yaml
--- 
instances: 
  - 
    arguments: 
      enable_bucket_stats: true
      enable_buckets: true
      enable_cluster_and_nodes: true
      hostname: localhost
      password: rdrdrdrd
      port: 8091
      query_port: 8093
      timeout: 30
      use_ssl: false
      username: rd_rd_rd
    command: all_data
    labels:
        PARTITION: RD
        REGION: RD1
        PRODUCT: RD
        ENVIRONMENT: RD
        PROJECT: RD
        COMPONENT: RD
integration_name: com.newrelic.couchbase
```

#### Error Message
```text
time="2020-01-27T11:49:14-08:00" level=debug msg="Integration command wrote to stderr" instance= integration=com.newrelic.couchbase prefix=config/couchbase stderr="[ERR] Error running collection for collector '': entity name and type are required when defining one\n" working-dir=/var/db/newrelic-infra/newrelic-integrations
```

#### OS

- [x] Ubuntu, versions 12.04, 14.04, and 16.04 (LTS  versions)

### PR DESCRIPTION
**Pull request by camdencheek**
#### Description of the changes

Fixes #32

### CODE REVIEW

File: src/entities/collector.go [hunk 1]
```diff
@@ -25,10 +25,15 @@ func SetClusterName(client *client.HTTPClient) error {
 	}
 
 	// if we couldn't get the cluster name (version 4.x) use the pool name instead
-	if clusterDetails.ClusterName == nil {
+	if clusterDetails.ClusterName != nil && *clusterDetails.ClusterName != "" {
+		ClusterName = *clusterDetails.ClusterName
+		return nil
+	} else if clusterDetails.PoolName != nil && *clusterDetails.ClusterName != "" {
```
Comment (icpenguins) [inline src/entities/collector.go]: else if clusterDetails.PoolName != nil && *clusterDetails.ClusterName != ""

Should it be?
else if clusterDetails.PoolName != nil && *clusterDetails.PoolName != ""

File: src/entities/collector.go [hunk 2]
```diff
@@ -25,10 +25,15 @@ func SetClusterName(client *client.HTTPClient) error {
...
 		ClusterName = *clusterDetails.PoolName
+		return nil
 	} else {
-		ClusterName = *clusterDetails.ClusterName
+		ClusterName = "default"
+		log.Warn("Neither name nor clusterName is set. Defaulting to 'default'")
```
Comment (icpenguins) [inline src/entities/collector.go]: Wordsmithing: The cluster name and pool name could not be found in the response. Using the value of 'default' for the cluster name.

File: src/entities/collector.go (hunk 1 above)
Comment (camdencheek) [reply src/entities/collector.go]: Good catch 👍

File: src/entities/collector.go [hunk 3]
```diff
@@ -25,10 +25,14 @@ func SetClusterName(client *client.HTTPClient) error {
...
-	if clusterDetails.ClusterName == nil {
+	if clusterDetails.ClusterName != nil && *clusterDetails.ClusterName != "" {
+		ClusterName = *clusterDetails.ClusterName
+		return nil
+	} else if clusterDetails.PoolName != nil && *clusterDetails.ClusterName != "" {
 		ClusterName = *clusterDetails.PoolName
+		return nil
 	} else {
-		ClusterName = *clusterDetails.ClusterName
+		ClusterName = "unknown"
```
Comment (icpenguins) [inline src/entities/collector.go]: `default`, similar to the log message?

File: src/entities/collector.go [hunk 4]
```diff
@@ -25,10 +25,15 @@ func SetClusterName(client *client.HTTPClient) error {
 	}
 
 	// if we couldn't get the cluster name (version 4.x) use the pool name instead
-	if clusterDetails.ClusterName == nil {
+	if clusterDetails.ClusterName != nil && *clusterDetails.ClusterName != "" {
+		ClusterName = *clusterDetails.ClusterName
+		return nil
```
Comment (icpenguins) [inline src/entities/collector.go]: return nil technically not needed as we fall out to return nil at the bottom

File: src/entities/collector.go [hunk 5]
```diff
@@ -25,10 +25,15 @@ func SetClusterName(client *client.HTTPClient) error {
...
+	} else if clusterDetails.PoolName != nil && *clusterDetails.PoolName != "" {
 		ClusterName = *clusterDetails.PoolName
+		return nil
```
Comment (icpenguins) [inline src/entities/collector.go]: return nil technically not needed as we fall out to return nil at the bottom

**Pull request merged**